guides/v2.2/ext-best-practices/extension-coding/security-performance-data-bp.md

```diff
@@ -23,16 +23,13 @@ Make sure that your Magento application does not directly use any PHP supergloba
   ```php
   $GLOBALS, $_SERVER, $_GET, $_POST, $_FILES, $_COOKIE, $_SESSION, $_REQUEST, $_ENV
   ```
+
 Instead use the [`Magento\Framework\HTTP\PhpEnvironment\Request`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/HTTP/PhpEnvironment/Request.php){:target="_blank"} wrapper class to safely access these values.
 
 ## Use the correct MySQL data types
 
 MySQL offers a range of numeric, string, and time data types. If you are storing a date, use a DATE or DATETIME field. Using an INTEGER or STRING can make SQL queries more complicated, if not impossible. It is often tempting to invent your own data formats; for example, storing serialized PHP objects in string. Database management may be easier, but MySQL will become a dumb data store and it may lead to problems later.
 
-## Get the correct data from the correct object
```

Comment:
Issue requested the following: ```The section should have correct links, use terms that are easier to understand and have a deeper dive text covering the topic.```

The section does seem like it's providing some useful information -- that is, you want to ensure that you are pulling the correct data from the Magento data base. For this example,  I am guessing that there is an order collection and a product collection which probably share the same data fields, and if you meant to get product data and got order data you would run into issues.   It seems like this could be stated more clearly with pointers to tutorial or other information that explained about pulling data from different collections in Magento.